apps/ledger-live-desktop/src/renderer/families/evm/EditTransaction/steps/StepFees.tsx

```diff
@@ -0,0 +1,148 @@
+import { getMainAccount } from "@ledgerhq/live-common/account/index";
+import { getFormattedFeeFields } from "@ledgerhq/live-common/families/evm/editTransaction/index";
+import React, { Fragment, memo } from "react";
+import { Trans } from "react-i18next";
+import { useSelector } from "react-redux";
+import Alert from "~/renderer/components/Alert";
+import Box from "~/renderer/components/Box";
+import Button from "~/renderer/components/Button";
+import BuyButton from "~/renderer/components/BuyButton";
+import CurrencyDownStatusAlert from "~/renderer/components/CurrencyDownStatusAlert";
+import logger from "~/renderer/logger";
+import SendAmountFields, { SendAmountFieldsProps } from "~/renderer/modals/Send/SendAmountFields";
+import { localeSelector } from "~/renderer/reducers/settings";
+import { TransactionErrorBanner } from "../components/TransactionErrorBanner";
+import { StepProps } from "../types";
+
+const StepFees = ({
+  account,
+  parentAccount,
+  transaction,
+  transactionToUpdate,
+  status,
+  bridgePending,
+  t,
+  onChangeTransaction,
+  updateTransaction,
+}: StepProps) => {
+  const mainAccount = getMainAccount(account, parentAccount);
+  const locale = useSelector(localeSelector);
+
+  // log fees info
+  logger.log(`transactionToUpdate.maxFeePerGas: ${transactionToUpdate.maxFeePerGas?.toFixed(0)}`);
+  logger.log(`transactionToUpdate.gasPrice: ${transactionToUpdate.gasPrice?.toFixed(0)}`);
+  logger.log(
+    `transactionToUpdate.maxPriorityFeePerGas: ${transactionToUpdate.maxPriorityFeePerGas?.toFixed(
+      0,
+    )}`,
+  );
+
+  const {
+    formattedFeeValue,
+    formattedMaxPriorityFeePerGas,
+    formattedMaxFeePerGas,
+    formattedGasPrice,
+  } = getFormattedFeeFields({ transaction: transactionToUpdate, mainAccount, locale });
+
+  return (
+    <Box flow={4}>
+      <CurrencyDownStatusAlert currencies={[mainAccount.currency]} />
+      <Fragment key={account.id}>
+        <SendAmountFields
+          account={mainAccount}
+          parentAccount={parentAccount}
+          status={status}
+          transaction={transaction}
+          /**
+           * Looks like TS does not like the fact that the arg of the functions
+           * are going from specific (EvmTransaction) to generic (Transaction).
+           */
+          onChange={onChangeTransaction as SendAmountFieldsProps["onChange"]}
```

Comment:
Why using `StepProps` in line 27 if you force a cast few lines later ? If StepProps don't have correct type you can create a type inside this file and just do:

```typescript
type StepFeesProps = Omit<StepProps, "onChangeTransaction" | "updateTransaction"> & {
 onChangeTransaction: SendAmountFieldsProps["onChange"];
 updateTransaction: SendAmountFieldsProps["updateTransaction"]
}
```

That way we sure to always pass the correct types from the parent component

Reply:
Indeed good idea

9d1ad3bdb03e7a362b935432cb4e727e06db6c9d